doc/content/integrations/cloud-integrations/withthegrid/_index.md

```diff
@@ -0,0 +1,374 @@
+    deviceTypeHashId,
+    deviceIdentifier,
+  };
+}
+
+function getJsonObject(data: Json | undefined): any {
+  if (typeof data === 'object' && data !== null && !Array.isArray(data)) {
+    return data;
+  } else {
+    throw new Error('Data must be a JSON object!');
+  }
+}
+```
+
+## Step 6: Configuring device type
+
+The device type that we have created will forward the payload to the report type (which we will create in step 8) where it will be parsed. The following code will validate the request and forward the actual payload generated by the sensor to the report parser.
+
+```tsx
+// define the hashId of the reportType
+const reportTypeHashId = 'eyr96d';
+
+/**
+ * Handle is the required function. It has two overloads, one for incoming HTTP requests and one for internal events
+ * When this function is called for an incoming HTTP requests, the function should return information about the response
+ * that should be returned.
+ */
+function handle(args: ArgumentsForIncomingRequestEvent, exec: Exec): IncomingRequestResponse;
+function handle(args: ArgumentsForEventWithoutResponse, exec: Exec): void;
+function handle(args: Arguments, exec: Exec): IncomingRequestResponse | void {
+  
+  if (args.event.type === 'incomingRequest') {
+    
+    const request = args.event.webRequest;
+
+    if (request.method === 'POST' && request.url.path === '/') {
+
+      if (request.body === undefined || request.body.type !== 'json') {
+        return {
+          statusCode: 400,
+          body: ['parameter_error', 'Body is not of type JSON'],
+        };
+      }
+
+      const payload = JSON.parse(JSON.stringify(request.body.data));
+      if (payload.join_accept !== undefined) {
+        return {
+          statusCode: 204,
+          body: ['TTN is registering!'],
+        };
+      }
+
+      // Set receive timestamp
+      let generatedAt = new Date();
+      generatedAt.setMilliseconds(0);
+
+      // pass parsing of the report to the right report type parser
+      exec.parseReport({
+        reportTypeHashId,
+        payload: JSON.stringify({
+          generatedAt,
+          payload: request.body.data,
+        }),
+      });
+           
+      return { statusCode: 204 };
+    }
+
+    return {
+      statusCode: 404,
+      body: ['not_found_error', 'Route cannot be found'],
+    };
+  }
+}
+```
+
+## Step 7: Configuring TTN and registering the device
+
+Now that the webhook and the event handler of Dragino - LWL02 are configured, we can set up  TTN project. 
+
+{{< figure src="Add_app.png" alt="TTN add app" >}}
+
+The second step is to add end device to the newly created application. For this you will need the AppEUI, DevEUI and the AppKey which is provided by the Dragino. After the device is registered, we can configure the application server by creating a new webhook. Set the webhook format to JSON and enter the Base URL listed in the Withthegrid webhook of TTN (step 2).
+
+{{< figure src="TTN_Webhook.png" alt="TTN set up webhook" >}}
```

Comment:
This image should be larger, I think you can crop the bottom part with all these message types enabled. Btw, do you really want to enable all message types? Isn't this just for uplink messages?